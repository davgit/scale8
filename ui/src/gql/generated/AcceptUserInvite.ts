/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { OrgAcceptInviteInput } from "./globalTypes";

// ====================================================
// GraphQL mutation operation: AcceptUserInvite
// ====================================================

export interface AcceptUserInvite {
  /**
   * @bound=Org
   * Accept a user invite to join an `Org`
   */
  acceptInvite: boolean;
}

export interface AcceptUserInviteVariables {
  orgAcceptInviteInput: OrgAcceptInviteInput;
}
