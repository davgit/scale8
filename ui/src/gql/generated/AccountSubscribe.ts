/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { AccountSubscribeInput } from "./globalTypes";

// ====================================================
// GraphQL mutation operation: AccountSubscribe
// ====================================================

export interface AccountSubscribe {
  /**
   * @bound=Org
   * Subscribe To TagManager
   * 
   * This will generate a new link that the user can follow to checkout if
   * it has no subscription yet, it will return null otherwise.
   */
  accountSubscribe: string | null;
}

export interface AccountSubscribeVariables {
  accountSubscribeInput?: AccountSubscribeInput | null;
}
