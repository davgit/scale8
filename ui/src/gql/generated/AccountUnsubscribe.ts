/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { AccountUnsubscribeInput } from "./globalTypes";

// ====================================================
// GraphQL mutation operation: AccountUnsubscribe
// ====================================================

export interface AccountUnsubscribe {
  /**
   * @bound=Org
   * Cancel subscription
   */
  accountUnsubscribe: boolean;
}

export interface AccountUnsubscribeVariables {
  accountUnsubscribeInput?: AccountUnsubscribeInput | null;
}
