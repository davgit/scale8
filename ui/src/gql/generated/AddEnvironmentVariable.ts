/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { EnvironmentVariableAddInput } from "./globalTypes";

// ====================================================
// GraphQL mutation operation: AddEnvironmentVariable
// ====================================================

export interface AddEnvironmentVariable {
  /**
   * @bound=Environment
   * Add an environment variable
   */
  addEnvironmentVariable: boolean;
}

export interface AddEnvironmentVariableVariables {
  environmentVariableAddInput: EnvironmentVariableAddInput;
}
