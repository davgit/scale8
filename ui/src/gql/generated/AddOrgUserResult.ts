/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { OrgAddUserInput } from "./globalTypes";

// ====================================================
// GraphQL mutation operation: AddOrgUserResult
// ====================================================

export interface AddOrgUserResult_addUser {
  __typename: "User";
  /**
   * `User` ID
   */
  id: string;
}

export interface AddOrgUserResult {
  /**
   * @bound=Org
   * Add a user to an `Org`
   */
  addUser: AddOrgUserResult_addUser;
}

export interface AddOrgUserResultVariables {
  orgAddUserInput: OrgAddUserInput;
}
