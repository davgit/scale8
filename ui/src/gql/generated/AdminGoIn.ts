/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { AdminAddMeToOrgInput } from "./globalTypes";

// ====================================================
// GraphQL mutation operation: AdminGoIn
// ====================================================

export interface AdminGoIn {
  /**
   * @bound=Org
   * A system admin can add himself to an `Org`, in order to offer support
   */
  adminAddMeToOrg: boolean;
}

export interface AdminGoInVariables {
  adminAddMeToOrgInput: AdminAddMeToOrgInput;
}
