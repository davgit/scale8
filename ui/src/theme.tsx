import { createTheme, adaptV4Theme } from '@mui/material/styles';
import { getProductSection, ProductSectionKey } from './containers/SectionsDetails';
import { grey } from '@mui/material/colors';

declare module '@mui/material/styles/createPalette' {
    interface Palette {
        adminColor: Palette['primary'];
        dataManagerColor: Palette['primary'];
        tagManagerColor: Palette['primary'];
        commonColor: Palette['primary'];
    }
    interface PaletteOptions {
        adminColor: PaletteOptions['primary'];
        dataManagerColor: PaletteOptions['primary'];
        tagManagerColor: PaletteOptions['primary'];
        commonColor: PaletteOptions['primary'];
    }
}

// A custom theme for this app
const theme = createTheme(
    adaptV4Theme({
        palette: {
            primary: {
                main: '#25302e',
            },
            secondary: {
                main: grey['700'],
            },
            error: {
                main: '#c63d51',
            },
            info: {
                main: '#444444',
            },
            success: {
                main: '#c6e54b',
            },
            background: {
                default: '#ffffff',
            },
            adminColor: {
                main: getProductSection(ProductSectionKey.admin).color,
            },
            dataManagerColor: {
                main: getProductSection(ProductSectionKey.dataManager).color,
            },
            tagManagerColor: {
                main: getProductSection(ProductSectionKey.tagManager).color,
            },
            commonColor: {
                main: getProductSection(ProductSectionKey.global).color,
            },
        },
        typography: {
            fontFamily: ['Source Sans Pro', 'Helvetica Neue', 'Arial', 'sans-serif'].join(','),
            fontSize: 14,
            body1: {
                fontSize: 14,
            },
        },
        overrides: {
            MuiTableCell: {
                head: {
                    fontWeight: 600,
                },
            },
            MuiSvgIcon: {
                root: {
                    fontSize: '1.25rem',
                },
            },
            MuiButton: {
                root: {
                    fontWeight: 600,
                },
            },
            MuiTooltip: {
                tooltip: {
                    fontSize: '.8rem',
                },
            },
            MuiDialogActions: {
                spacing: {
                    '& > :not(:first-child)': {
                        marginLeft: '16px',
                    },
                },
            },
        },
    }),
);

export default theme;
