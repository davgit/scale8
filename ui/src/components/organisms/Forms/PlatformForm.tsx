import { FC, useState } from 'react';
import ControlledTextInput from '../../atoms/ControlledInputs/ControlledTextInput';
import ControlledTextAreaInput from '../../atoms/ControlledInputs/ControlledTextAreaInput';
import { PlatformFormProps } from '../../../types/props/forms/PlatformFormProps';
import DrawerFormLayout from '../../molecules/DrawerFormLayout';
import ImageUpload from '../../atoms/ImageUpload';
import ControlledSelect from '../../atoms/ControlledInputs/ControlledSelect';
import { PlatformType } from '../../../gql/generated/globalTypes';

const PlatformForm: FC<PlatformFormProps> = (props: PlatformFormProps) => {
    const [imageLoading, setImageLoading] = useState(false);

    return (
        <DrawerFormLayout {...props} submitDisable={props.isSubmitting || imageLoading}>
            <ControlledTextInput
                name="name"
                label="Name"
                formProps={props}
                className="DialogFormField"
                required
                autoFocus
            />
            <ControlledTextAreaInput
                name="description"
                label="Description"
                formProps={props}
                className="DialogFormField"
                required
            />
            {!props.isUpdate && (
                <ControlledSelect
                    className="DialogFormField"
                    label="Type"
                    name="type"
                    values={[
                        {
                            key: PlatformType.CUSTOM,
                            text: 'Custom',
                        },
                        {
                            key: PlatformType.TEMPLATED,
                            text: 'Templated',
                        },
                    ]}
                    formProps={props}
                    required
                />
            )}
            <ImageUpload
                imageLoading={imageLoading}
                setImageLoading={setImageLoading}
                initialUrl={props.initialUrl}
                removeImage={props.removeImage}
                handleImageUpload={props.handleImageUpload}
                label={'Platform Logo'}
            />
        </DrawerFormLayout>
    );
};

export default PlatformForm;
