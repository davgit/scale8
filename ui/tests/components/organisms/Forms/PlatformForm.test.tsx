import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import theme from '../../../../src/theme';
import { FC } from 'react';
import { useMockFormProps } from '../../../componentsMocks/useMockFormProps';
import PlatformForm from '../../../../src/components/organisms/Forms/PlatformForm';
import { PlatformValues } from '../../../../src/utils/forms/PlatformFormUtils';
import { PlatformType } from '../../../../src/gql/generated/globalTypes';

const MockForm: FC = () => {
    const formProps = useMockFormProps<PlatformValues>({
        name: '',
        description: '',
        fileName: '',
        type: PlatformType.TEMPLATED,
    });

    return (
        <PlatformForm
            {...formProps}
            removeImage={() => {
                /*Idle*/
            }}
            handleImageUpload={() => {
                /*Idle*/
            }}
        />
    );
};

describe('PlatformForm', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it('form render unchanged', () => {
        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <MockForm />
            </ThemeProvider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
