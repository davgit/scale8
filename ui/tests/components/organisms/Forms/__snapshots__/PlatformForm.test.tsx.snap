// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlatformForm form render unchanged 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-s92abg"
  >
    <div
      class="MuiBox-root css-70qvj9"
    >
      <div
        class="MuiBox-root css-1isemmb"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-uix0gv"
      >
        mock form
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiBox-root css-yn9cjk"
    >
      <div
        class="MuiBox-root css-1676va5"
      >
        <form
          class="MuiBox-root css-mjj15b"
        >
          <div
            class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error Mui-focused Mui-required css-1oyovms-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
            >
              Name
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk Mui-error css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-error Mui-focused MuiInputBase-formControl css-1ejkwvn-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-invalid="true"
                autocomplete="off"
                class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                name="name"
                required=""
                type="text"
                value=""
              />
            </div>
            <p
              class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium Mui-focused Mui-required css-1sbojr7-MuiFormHelperText-root"
            >
              Required value
            </p>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error Mui-required css-ip4mx2-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
            >
              Description
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk Mui-error css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl MuiInputBase-multiline css-1yxqjki-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <textarea
                aria-invalid="true"
                autocomplete="off"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                name="description"
                required=""
                style="height: 0px; overflow: hidden;"
              />
              <textarea
                aria-hidden="true"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                tabindex="-1"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-dhh0px"
                >
                  <span>
                    Description *
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained Mui-required css-pr0kdb-MuiFormHelperText-root"
            >
              Required value
            </p>
          </div>
          <div
            class="MuiFormControl-root DialogFormField css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1oyovms-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
            >
              Type
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-1ejkwvn-MuiInputBase-root-MuiInput-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-type"
                class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                id="mui-component-select-type"
                role="button"
                tabindex="0"
              >
                Templated
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="type"
                required=""
                tabindex="-1"
                value="TEMPLATED"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-yqkyv6-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <label
            class="MuiBox-root css-26oc1r"
          >
            Platform Logo
          </label>
          <input
            accept="image/jpeg,image/png"
            class="MuiBox-root css-1hyfx7x"
            id="icon-button-file"
            type="file"
          />
          <label
            for="icon-button-file"
          >
            <div
              class="MuiBox-root css-1a8nwmw"
            >
              <div
                class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault css-r4bwd2-MuiAvatar-root"
              >
                Select Logo
              </div>
              <small
                class="MuiBox-root css-wpfbeq"
              >
                (Square JPG or PNG, Max 32Kb)
              </small>
              <button
                aria-label="delete"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1vgxi9x-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-17ii7u1-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </label>
          <div
            class="MuiBox-root css-wpygli"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-xcbv7o-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
